tests/s3/test_async_delete.py

```diff
@@ -0,0 +1,446 @@
+from libs.s3.s3_test_lib import S3TestLib
+from libs.s3.s3_versioning_test_lib import S3VersioningTestLib
+from libs.s3.s3_versioning_common_test_lib import check_get_head_object_version
+from libs.s3.s3_versioning_common_test_lib import check_list_objects
+from libs.s3.s3_versioning_common_test_lib import check_list_object_versions
+from libs.s3.s3_versioning_common_test_lib import delete_version
+from libs.s3.s3_versioning_common_test_lib import download_and_check
+from libs.s3.s3_versioning_common_test_lib import empty_versioned_bucket
+from libs.s3.s3_versioning_common_test_lib import upload_version
+
+
+class TestAsyncDelete:
+    """Test CORTX RGW Async Delete"""
+
+    @classmethod
+    def setup_class(cls):
+        """
+        Function will be invoked prior to all the test cases.
+        """
+        cls.log = logging.getLogger(__name__)
+        cls.log.info("STARTED: Setup test suite level operations.")
+        cls.s3_test_obj = S3TestLib(endpoint_url=S3_CFG["s3_url"])
+        cls.s3_ver_obj = S3VersioningTestLib(endpoint_url=S3_CFG["s3_url"])
+        cls.fixed_size_fpaths = []
+        cls.total_fixed_file_size = 0
+        cls.test_dir_path = os.path.join(TEST_DATA_FOLDER, "TestAsyncDelete")
+        if not path_exists(cls.test_dir_path):
+            make_dirs(cls.test_dir_path)
+            cls.log.info("Created path: %s", cls.test_dir_path)
+        cls.file_prefix = "async_del"
+        cls.object_prefix = "obj_async_del"
+        cls.bucket_prefix = "buck-async-del"
+        for file_size in DEL_CFG["fixed_obj_size"]:
+            file_name = f"{cls.file_prefix}{time.perf_counter_ns()}"
+            file_path = os.path.join(cls.test_dir_path, file_name)
+            create_file(file_path, file_size, "/dev/urandom")
+            cls.log.info("Created file: %s", file_path)
+            cls.fixed_size_fpaths.append(file_path)
+            cls.total_fixed_file_size += os.path.getsize(file_path)
+        cls.log.info("ENDED: Setup test suite level operations.")
+
+    @classmethod
+    def teardown_class(cls):
+        """
+        Function will be invoked after completion of all the test cases.
+        """
+        cls.log.info("STARTED: teardown test suite operations.")
+        if path_exists(cls.test_dir_path):
+            remove_dirs(cls.test_dir_path)
+        cls.log.info("Cleanup test directory: %s", cls.test_dir_path)
+        cls.log.info("ENDED: teardown test suite operations.")
+
+    # pylint: disable=attribute-defined-outside-init
+    def setup_method(self):
+        """
+        Function will perform prerequisite test steps prior to each test case.
+        """
+        self.log = logging.getLogger(__name__)
+        self.log.info("STARTED: Setup test method operations")
+        self.bucket_name = f"{self.bucket_prefix}-{time.perf_counter_ns()}"
+        self.obj_names = []
+        self.upload_paths = []
+        for _ in range(len(self.fixed_size_fpaths)):
```

Comment:
NO need of `range(len` if `fixed_size_fpaths` is list.

Reply:
done